### ISSUE TITLE
Add private device display names

### ISSUE THREAD

**Issue by tristankretzer** [matrix-org/matrix-doc#1826]
It would be nice to have private display names for devices (i. e. display names only visible to the user). With that you would be able to organize your devices without leaking information to the public (e. g. I have a certain way to identify my devices but I do not want these identifiers to be public).

**Comment by ara4n** [comment]
+1. it'd also be useful to optionally view the geolocation and other info about your own devices.
however, we should probably sign this info to stop malicious server admins confusing users by lying about this data.

**Comment by bertob** [comment]
I think this issue is bigger than just having a separate private device name.

The current session list in the client settings is not very useful for finding specific sessions, because it only includes a name (which is not unique and different clients use differently), and the last seen timestamp.

In order to actually be able to differentiate devices the hardware, OS, client version, and location would be good to have when possible. A single private name string is not sufficient for that, and a few metadata fields would be needed.

Here's what Telegram does, for example:

![image](https://user-images.githubusercontent.com/1908896/53339505-dc4afe00-3906-11e9-84d5-a8832e216d84.png)

**Comment by ara4n** [comment]
agreed. but we would definitely need to sign such data by the client that provides it (and track whether we trust said client) otherwise it makes it trivial for a malicious server admin to fake a malicious device
